engine/src/main/java/org/pentaho/di/base/AbstractMeta.java

```diff
@@ -263,6 +260,8 @@ public void setAlwaysShowRunOptions( boolean alwaysShowRunOptions ) {
       new DelegatingSharedObjectsIO( bowl.getSharedObjectsIO(), localSharedObjects );
   protected DatabaseManagementInterface readDbManager = new PassthroughDbConnectionManager( combinedSharedObjects );
 
+  protected SlaveServerManagementInterface readSlaveServerManager = new PassthroughSlaveServerManager( combinedSharedObjects );
+
   protected void initializeLocalDatabases() {
```

Comment:
I just realized that this needs to be updated to reset all the shared object managers, not just databases. 

See the comment for why. :)

Reply:
line 266-267. 
```// NOTE: this has to assign new objects, not just clear existing ones, because it is used in clone(), and
    // updating the original objects will update the source of the clone.```